tests/unit/s2n_server_extensions_test.c

```diff
@@ -638,6 +658,11 @@ int main(int argc, char **argv)
                 EXPECT_SUCCESS(s2n_connection_allow_all_response_extensions(client_conn));
                 client_conn->actual_protocol_version = S2N_TLS13;
 
+                /* Fake receiving the key share extension */
+                s2n_extension_type_id key_share_id;
+                s2n_extension_supported_iana_value_to_id(s2n_server_key_share_extension.iana_value, &key_share_id);
+                S2N_CBIT_SET(client_conn->extension_responses_received, key_share_id);
```

Comment:
You're doing this a lot -- worth a testlib method? We also already have [one that sets all requests](https://github.com/aws/s2n-tls/blob/a7b97a0dcff52c3efec303a0204c75200963777f/tests/testlib/s2n_testlib.h#L61). Maybe add "s2n_connection_allow_response_extension(conn, iana_value)"

Reply:
I added `s2n_connection_allow_response_extension` and `s2n_connection_allow_request_extension` in 0149747